breathecode/certificate/tests/urls/tests_certificate_cohort_id.py

```diff
@@ -0,0 +1,588 @@
+        response = self.client.post(url, format='json')
+        json = response.json()
+        expected = {
+            'detail': 'no-user-with-student-role',
+            'status_code': 400
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(self.all_user_specialty_dict(), [])
+        
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_generate_certificate_with_role_student_with_cohort_user_without_syllabus(self):
+        """Test /certificate/cohort/id """
+        """ No syllabus """
+        self.headers(academy=1)
+        self.generate_models(authenticate=True, cohort=True, user=True,
+            profile_academy=True, capability='crud_certificate', role='STUDENT', 
+            cohort_user=True, cohort_stage="ENDED")
+
+        url = reverse_lazy('certificate:certificate_cohort', kwargs={'cohort_id': 1})
+        response = self.client.post(url, format='json')
+        json = response.json()
+        expected = {
+            'detail': "cohort-has-no-syllabus-assigned",
+            'status_code': 400
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(self.all_user_specialty_dict(), [])
+    
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_generate_certificate_with_role_student_with_syllabus_without_specialty(self):
+        """Test /certificate/cohort/id """
+        """ No specialty """
+        self.headers(academy=1)
+        self.generate_models(authenticate=True, cohort=True, user=True,
+            profile_academy=True, capability='crud_certificate', role='STUDENT', 
+            cohort_user=True, syllabus=True, cohort_stage="ENDED")
+
+        url = reverse_lazy('certificate:certificate_cohort', kwargs={'cohort_id': 1})
+        response = self.client.post(url, format='json')
+        json = response.json()
+        expected = {
+            'detail': "specialty-has-no-certificate-assigned",
+            'status_code': 400
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(self.all_user_specialty_dict(), [])
+    
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_generate_certificate_(self):
```

Comment:
This test is the same that `test_generate_certificate_with_role_student_with_syllabus_without_specialty`, remove the test `test_generate_certificate_`